drivers: display: gc9x01: convert to MIPI DBI API

danieldegrasse opened the pull request:
Convert galaxycore GC9X01 to MIPI DBI API. In tree boards and tests
using this display have also had their devicetrees updated to use the
new MIPI DBI SPI emulated device.

danieldegrasse committed:
doc: releases: migration-guide: add migration note for GC9X01 display

Add migration note to GC9X01 display, which has been converted to the
MIPI DBI API.

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

danieldegrasse commented:
@nicogou, would you be willing to test this change? I don't have hardware to verify it, but I'd like to move this display to the new MIPI DBI API

kartben commented:
@joelguittet might also be able to test on `esp32s3_touch_lcd_1_28_esp32s3` which he contributed

joelguittet commented on boards/shields/seeed_xiao_round_display/seeed_xiao_round_display.overlay:
```diff
@@ -27,6 +27,26 @@
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		gc9a01_xiao_round_display: gc9a01@0 {
```
Why moving this in the device tree ? This is more difficult to review and identify the changes ....

danieldegrasse replied:
This display uses a 4 wire SPI interface for commands and data, which means it can be supported by the MIPI DBI API. For more information on this API, you can look at the initial RFC: #64587. The main idea of the API is to better support display hardware that explicitly accelerates SPI 3/4 wire mode displays. This change will allow gc9x01 displays to operate with LCD hardware using the MIPI DBI API, as well as existing SPI controllers

For cases where a SPI peripheral is used to implement the 3/4 wire mode support, we need to use an emulated MIPI DBI driver that uses the SPI API. This driver is implemented by the `zephyr,mipi-dbi-spi` compatible, so we need to declare the display as a child of this driver's node.

Hope this helps explain the rationale for this change. Would you be willing to test it?

joelguittet commented on tests/drivers/build_all/display/app.overlay:
```diff
@@ -81,34 +91,26 @@
-			};
 
-			test_led_strip_0: lpd8806@2 {
+			test_led_strip_0: lpd8806@0 {
```
Why this change ?

joelguittet replied:
I guess a separated commit will be better ?

danieldegrasse replied:
Sure, I'll split this into another commit

danieldegrasse commented:
> Yes I can test on the Waveshare board too.
Just for the context please, what is the benefit of the MIPI DBI API compared to the spi interface ?
I just understand it is more complicated in the declaration of the device tree...

Thanks for testing it! I added a comment inline- I hope this helps explain the reasoning behind the MIPI DBI API

danieldegrasse committed:
drivers: display: gc9x01: convert to MIPI DBI API

Convert galaxycore GC9X01 to MIPI DBI API. In tree boards and tests
using this display have also had their devicetrees updated to use the
new MIPI DBI SPI emulated device.

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

danieldegrasse committed:
tests: drivers: build_all: display: clean up reg addresses

Clean up register addresses for devices defined as children of the dummy
SPI bus, so that they start from zero. This ensures all SPI chip select
gpio indices will be used

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

jfischer-no reviewed (dismissed):
LGTM

danieldegrasse commented:
Rebased to clear conflicts (this likely won't be the last time, #73812 and #73946 also touch the display build all test)

kartben commented:
> Rebased to clear conflicts (this likely won't be the last time

Correct :)

danieldegrasse commented:
> > Rebased to clear conflicts (this likely won't be the last time
> 
> Correct :)

Rebased again- #73812 and this both touch the same build all test, so if one merges I'll update the other.

danieldegrasse commented:
Final rebase- @jfischer-no if you have a chance to refresh +1 it would be appreciated (but I know it is late, so no worries!)
